frontend/src/app/login-form/login-form.component.ts

```diff
@@ -4,25 +4,30 @@ import { HttpService, LoginResponse } from './../services/http.service';
 @Component({
   selector: 'app-login-form',
   templateUrl: './login-form.component.html',
-  styleUrls: ['./login-form.component.scss']
+  styleUrls: ['./login-form.component.scss'],
 })
 export class LoginFormComponent {
   hide = true;
   token: string = null;
+  error = null;
+  username = '';
+  password = '';
 
   constructor(private httpService: HttpService) {}
 
-  loginHandler(username: string, password: string) {
-    this.httpService.getToken(username, password).subscribe({
+  loginHandler() {
+    this.httpService.getToken(this.username, this.password).subscribe({
       next: (data: LoginResponse) => {
+        this.error = null;
         this.token = data.token;
-        console.log(this.token);
       },
-      error: (err) => { console.log(err.error); },
+      error: (err) => {
+        this.error = err.error;
+        this.token = null;
+      },
     });
```

Comment:
Can be also:
```ts
.subscribe(
  data => { // on next callback },
  err => { // on error callback }
)
```

Reply:
Do you mean to extract these blocks into separate functions?

Reply:
Nah, just want to signalise that there are two different options to pass callbacks:
```ts
subscribe({ next: data => { //... }, error: err => { //... } })
```
vs.
```ts
subscribe(data => { ..///}, err => { //... })
```

Reply:
Looks better without additional pair of brackets ;)